elements/nuxeo-browser/nuxeo-breadcrumb.js

```diff
@@ -14,192 +14,252 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-      }
+    static get is() {
+      return 'nuxeo-breadcrumb';
+    }
 
-      .right-ellipsis {
-        direction: ltr;
-        text-align: right;
-      }
+    static get properties() {
+      return {
+        document: {
+          type: Object,
+          observer: '_setBreadcrumbElements',
+        },
+        _breadcrumb: {
+          type: Object,
+          value: {},
+        },
+      };
+    }
 
-      @media (max-width: 1024px) {
-        .current-icon {
-          display: none;
-        }
-      }
-    </style>
-
-    <nuxeo-connection id="nxcon" url="{{url}}"></nuxeo-connection>
-
-    <div class="breadcrumb">
-      <div class="current-icon">
-        <iron-icon src="[[_icon(document, url)]]"></iron-icon>
-      </div>
-      <div class="doc-path">
-        <a href$="[[urlFor('browse', document.path)]]" class="current breadcrumb-item breadcrumb-item-current">
-          [[_title(document)]]
-        </a>
-        <div class$="ancestors [[_ellipsisDirection()]]">
-          <template is="dom-repeat" items="[[_breadcrumb(document)]]">
-            <span class$="breadcrumb-item [[_computeCssClass(index, document)]]">
-              <a href$="[[urlFor('browse', item.path)]]">
-                <span class="breadcrumb-item-title">[[item.title]]&lrm;</span>
-              </a>
-              <span class="breadcrumb-divider">&gt;</span>
-            </span>
-          </template>
-        </div>
-      </div>
-    </div>
-  `,
-
-  is: 'nuxeo-breadcrumb',
-  behaviors: [RoutingBehavior, I18nBehavior],
-
-  properties: {
-    document: {
-      type: Object,
-    },
-  },
-
-  _breadcrumb() {
-    if (this._enrichers) {
-      return this._enrichers.breadcrumb.entries.slice(0, this._enrichers.breadcrumb.entries.length - 1);
+    connectedCallback() {
+      super.connectedCallback();
+      this.addEventListener('iron-resize', this._resize);
     }
-  },
 
-  _computeCssClass(index) {
-    if (this._enrichers) {
-      if (index === this._enrichers.breadcrumb.entries.length - 1) {
-        return 'breadcrumb-item-current';
-      }
-      if (index === this._enrichers.breadcrumb.entries.length - 2) {
-        return 'breadcrumb-item-parent';
-      }
-      if (index === this._enrichers.breadcrumb.entries.length - 3) {
-        return 'breadcrumb-item-grand-parent';
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      this.removeEventListener('iron-resize', this._resize);
+    }
+
+    get ancestors() {
+      const ancestors = this.shadowRoot.getElementById('ancestors');
+      return ancestors;
+    }
+
+    get _enrichers() {
+      return this.document && this.document.contextParameters;
+    }
+
+    get contentWidth() {
+      return Array.from(this.ancestors.children).reduce((sum, current) => sum + current.offsetWidth, 0);
+    }
+
+    _getBreadcrumbElements() {
+      if (this._enrichers) {
+        return this._enrichers.breadcrumb.entries.slice(0, this._enrichers.breadcrumb.entries.length - 1);
```

Comment:
nit: could be a one-liner:
```suggestion
      return this._enrichers && this._enrichers.breadcrumb.entries.slice(0, this._enrichers.breadcrumb.entries.length - 1);
```